src/renderer/components/opds/EntryPublicationList.tsx

```diff
@@ -9,21 +9,28 @@ import * as qs from "query-string";
 import * as React from "react";
 import { connect } from "react-redux";
 import { RouteComponentProps, withRouter } from "react-router-dom";
-import { OpdsPublicationView } from "readium-desktop/common/views/opds";
-import { PublicationView } from "readium-desktop/common/views/publication";
+import { OpdsPublicationView, OpdsResultPageInfos, OpdsResultUrls } from "readium-desktop/common/views/opds";
 import { DisplayType } from "readium-desktop/renderer/components/opds/Header";
 import GridView from "readium-desktop/renderer/components/utils/GridView";
 import ListView from "readium-desktop/renderer/components/utils/ListView";
 import Loader from "readium-desktop/renderer/components/utils/Loader";
 import { RootState } from "readium-desktop/renderer/redux/states";
 
+import PageNavigation from "./PageNavigation";
+
 interface IProps extends RouteComponentProps {
-    publications: OpdsPublicationView[] | undefined;
```

Comment:
if you test publications above in render, then, the publications can be typed in undefined, no ?

Reply:
I didn't do that on purpose, it's an mistake when I resolved the conflicts.
But is it really usefull to type it with undefined ? A type takes undefined by default, no ?

Reply:
not really, it's more clear to strictly typing our code.
You can see the rule strictNullChecks
https://www.tsmean.com/articles/learn-typescript/strict-null-checks-best-practice/

I'm not a specialist, it's my opinion

Reply:
As you know, I am very much in support of removing `any` (@panaC refactorings are incrementally addressing this problem). Also, I am in favour of eventually introducing proper static / compile-time verification of `undefined` versus `null` usages (i.e. `strictNullChecks` TypeScript option).

Reminder: in Javascript `typeof null === "object"` vs. `typeof undefined === "undefined"`, and in TypeScript both `null` and `undefined` are assignable to `any` (and `undefined` is assignable to `void`).

These distinctions are important because the `readium-desktop` codebase (including the R2 "js" components / NPM package dependencies) must be capable of differentiating `null` object properties, vs. `undefined` / non-existing ones, notably when parsing arbitrary / varied JSON formats. Both 'null' and `undefined` are "falsy", but the our codebase sometimes need to check for a specific type.

I had started to explore the impact of adding `strictNullChecks` support in the `readium-desktop` codebase (see this WIP PR: https://github.com/readium/readium-desktop/pull/475/files ), but it quickly became obvious that this would need to be a gradual refactoring effort.

We started this effort a few weeks ago, and now at every PR we aim to implement stricter typing. Unfortunately the `strictNullChecks` compiler option is not activated yet (otherwise the entire codebase breaks) ... so we must rely on "common-sense", rather than TypeScript error messages.

Here is how the Visual Studio Code development team iterated over a much larger codebase than `readium-desktop`:
https://code.visualstudio.com/blogs/2019/05/23/strict-null

Documentation:
https://www.typescriptlang.org/docs/handbook/compiler-options.html

Reply:
This comment has been addressed: https://github.com/readium/readium-desktop/pull/733/commits/2fcf478637b8b6d3ef2462ae4a5739c36cad5aa7

Reply:
Thank you Daniel, perfect explanation

Reply:
`strictPropertyInitialization` is a good one too ... but one thing at a time ;)